swarm/src/behaviour.rs

```diff
@@ -40,6 +41,47 @@ pub(crate) type THandlerInEvent<THandler> =
 pub(crate) type THandlerOutEvent<THandler> =
     <<THandler as IntoConnectionHandler>::Handler as ConnectionHandler>::OutEvent;
 
+/// Implementation of [`NetworkBehaviour`] that combines two underlying implementations.
+#[cfg(feature = "macros")]
+#[derive(Debug, Default, libp2p_swarm_derive::NetworkBehaviour)]
+#[behaviour(prelude = "crate::derive_prelude")]
+pub struct And<A, B> {
+    first: A,
+    second: B,
+}
```

Comment:
`NetworkBehaviour` combinators like `Toggle` and `Either` are in their own file. I would suggest doing the same for `And`.

Reply:
Thanks, Addressed!